File: c/5-matrices/above-main-diagonal/main.c
/*
Write a program that creates a 4x4 matrix, and asks the user to enter numbers.
Print the values ​​that are above the main diagonal.
*/

#include <stdio.h>
#include <stdlib.h>

int main()
{

    int mat[4][4], i, j, number;

    printf("Enter matrix numbers:\n");
    for (i = 0; i < 4; i++) {
        for (j = 0; j < 4; j++) {
            scanf("%d", &number);
            mat[i][j] = number;
        }
    }

    for (i = 0; i < 4; i++) {
        for (j = 0; j < 4; j++) {
            if (j > i) {
                printf("%d  ", mat[i][j]);
            }
            else {
                printf("    ");
            }
        }
        printf("\n");
    }

    return 0;
}

File: java/5-matrices/above-main-diagonal/src/Main.java
/*
Write a program that creates a 4x4 matrix, and asks the user to enter numbers.
Print the values that are above the main diagonal.
*/

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int[][] mat = new int[4][4];

        System.out.println("Enter matrix numbers:");
        for (int i = 0; i < 4; i++) {
            for (int j = 0; j < 4; j++) {
                int number = sc.nextInt();
                mat[i][j] = number;
            }
        }

        System.out.println("\nAbove main diagonal:");
        for (int i = 0; i < 4; i++) {
            for (int j = 0; j < 4; j++) {
                if (j > i) {
                    System.out.print(mat[i][j] + "    ");
                }
                else {
                    System.out.print("    ");
                }
            }
            System.out.println();
        }
        
    }
}